### ISSUE TITLE
make install to a --chpl-home that doesn't exist, fails

### ISSUE THREAD

**Issue by tonycurtis** [chapel-lang/chapel#23264]
### Summary of Problem

```
$ ./configure --chpl-home=/path/to/somewhere
...
...
$ make install
Making the modules...
make[2]: Nothing to be done for 'default'.
Detecting CHPL_HOME as /lustre/home/arcurtis/chapel/chapel-1.31.0
Exiting: Installation dest path '/lustre/home/arcurtis/opt/chapel/1.31.0' does not exist
```

Of course it doesn't exist, that's why I am installing it there!

**Comment by bradcray** [comment]
@tonycurtis : Interesting...  When I try specifying a path that doesn't exist, I get:

```bash
$ ./configure --chpl-home=~/does/not/exist
Prefix directory does not exist, creating directory.
```

and it does in fact seem to create the directory:

```bash
$ tree does
does
└── not
    └── exist

3 directories, 0 files
```

Or, if I try to create a directory that I don't have permissions to make, I get a different error:

```bash
$ ./configure --chpl-home=/does/not/exist
Prefix directory does not exist, creating directory.
mkdir: /does: Read-only file system
./configure: line 99: cd: /does/not/exist: No such file or directory
Error: Could not cd into prefix directory ''
       ... perhaps it does not exist?
```

In the `...` output you've elided, do you not see a similar message to either of these?


Also, does the path that `make install` complains about in this line:

```
Exiting: Installation dest path '/lustre/home/arcurtis/opt/chapel/1.31.0' does not exist
```

differ from the one you're specifying to `./configure`:

```bash
$ ./configure --chpl-home=/path/to/somewhere
```

as in the output you provided?  Or do they match?

**Comment by tonycurtis** [comment]
I think I originally did a configure to a directory that did exist, then I rm-rf'ed it for a completely clean install

**Comment by tonycurtis** [comment]
My target directory is under my $HOME, so there are no permission problems

**Comment by tonycurtis** [comment]
Recreatrion:

```
$ rm -rf ~/opt/chapel/1.310
$ make install
Making the modules...
make[2]: Nothing to be done for 'default'.
Detecting CHPL_HOME as /lustre/home/arcurtis/chapel/chapel-1.31.0
Exiting: Installation dest path '/lustre/home/arcurtis/opt/chapel/1.31.0' does not exist
make: *** [Makefile:230: install] Error 255
```

but then

```
$ mkdir  -p /lustre/home/arcurtis/opt/chapel/1.31.0
$ make install
```

goes as expected

**Comment by bradcray** [comment]
> then I rm-rf'ed it for a completely clean install

You're saying that this was after the `./configure` command, but before the `make install`?  OK, yeah, doing that, I can reproduce this.

I suspect that the current behavior is as its developer intended ("`./configure` creates the directory if necessary; `make install` uses the directory that `./configure` created").  Do you want us to treat this issue as a feature request that `make install` should re-create the directory if it has disappeared since `./configure` created it?

**Comment by tonycurtis** [comment]
Since you are using a `./configure` script this makes me (and probably other people) think GNU autoconf, but then your treatment of `--chpl-home` and `--prefix` are confusing.  Your infrastructure should definitely try to create the installation path.

**Comment by bradcray** [comment]
> Your infrastructure should definitely try to create the installation path.

Well, I mean, arguably it does since `./configure` is part of our infrastructure...

But doing a quick experiment, I see that autoconf-based `./configure` + `make install` doesn't create the directory at the configure stage, but at the (beginning of) the `make install` stage (I'd never stopped between the commands to check).  So we could certainly move that logic to be part of `make install` to match that more closely.

**Comment by mppf** [comment]
> But doing a quick experiment, I see that autoconf-based `./configure` + `make install` doesn't create the directory at the configure stage, but at the (beginning of) the `make install` stage (I'd never stopped between the commands to check). So we could certainly move that logic to be part of `make install` to match that more closely.

Right, and it is arguably wrong for us to do the `mkdir` in the configure step. (I don't know why we did that; there might be a reason...). The issue is that `./configure` (as well as `make`) should be run by a regular user, but `make install` should be run by `root` if it's installing to some system-wide path e.g. `/opt/chapel`). In that usage, `./configure` won't have the permissions needed to create the installation directory; but `make install` will.
